pkg/kubelet/apis/config/types.go

```diff
@@ -305,6 +305,11 @@ type KubeletConfiguration struct {
 	// This flag accepts a list of options. Acceptable options are `pods`, `system-reserved` & `kube-reserved`.
 	// Refer to [Node Allocatable](https://git.k8s.io/community/contributors/design-proposals/node/node-allocatable.md) doc for more information.
 	EnforceNodeAllocatable []string
+	// This flag specifies the interval of reporting volume processing failure event to pod.
+	// E.g. when configuring a wrong FlexVolume name in pod, "no volume plugin matched" error would be thrown out.
+	// Volume manager can generate a lot of similar errors because of its desired state of world loop.
+	// User can config the interval of error events reported to pod by this kubelet flag.
+	ProcessVolumeFailureEventInterval metav1.Duration
```

Comment:
What meaningful decision are operators making here about how the kubelet should be configured?  We generally avoid adding "tuning" variables to the kubelet and strongly prefer just making the kubelet do the right thing by default.

Reply:
I made the suggestion of making the throttling period configurable, because this event has the chance to be high frequency and we might not get the interval right and may want a way to turn it off in production.  But I agree this is a lot of instrumentation for this one message.  Is there an alternative way we could try to propagate this error up?

Reply:
Maybe go with something super conservative to begin with?  1 event per minute?  That to me is better than a new configuration.

Reply:
I am also curious why the errors could be so frequent.  If we fail to mount a volume, how often do we retry?  I realize shorter periods mean less latency, but we shouldn't encounter failures that often, right?  The kubelet's housekeeping sync loop is every 30 seconds, which seems to work fine as a retry rate, and is infrequent enough that we can omit events for errors we encounter.  Am I missing something?

Reply:
This error is generated as part of the volume manager desired world populator loop, which is separate from kubelet's main sync loop.  It runs every [100ms](https://sourcegraph.com/github.com/kubernetes/kubernetes/-/blob/pkg/kubelet/volumemanager/volume_manager.go#L61)